apps/user-profile/src/app/pages/settings-page/settings-authorization/settings-authentication.component.ts

```diff
@@ -165,65 +152,6 @@ export class SettingsAuthenticationComponent implements OnInit, AfterViewInit {
       }
     });
   }
-
-  private loadMFA() {
-    this.loading = true;
-    const enforceMfaAttributeName = this.store.get('mfa', 'enforce_mfa_attribute');
-    const tokensAttributeName = this.store.get('mfa', 'tokens_attribute');
-    this.attributesManagerService.getUserAttributeByName(this.store.getPerunPrincipal().userId, enforceMfaAttributeName).subscribe(attr => {
-      if (sessionStorage.getItem('mfa_route')) {
-        sessionStorage.removeItem('mfa_route');
-        this.mfaService.enableMfa(!attr || !attr.value, this.idToken).subscribe(() => {
-          // Should I do here something else than loadMFA()?
-          this.loadMFA();
-        }, () => this.loadMFA());
-      } else {
-        if (!attr) {
-          this.attributesManagerService.getAttributeDefinitionByName(enforceMfaAttributeName).subscribe(att => {
-            this.mfaAtt = att as Attribute;
-          });
-        } else {
-          this.mfaAtt = attr;
-
-          if (this.toggle) {
-            if (this.mfaAtt.value) {
-              this.toggle.toggle();
-            }
-
-            this.toggle.change.subscribe(() => {
-              this.reAuthenticate();
-            });
-          }
-
-          this.attributesManagerService.getUserAttributeByName(this.store.getPerunPrincipal().userId, tokensAttributeName).subscribe(mfaTokenAttr => {
-            if (!mfaTokenAttr) {
-              this.attributesManagerService.getAttributeDefinitionByName(tokensAttributeName).subscribe(att => {
-                mfaTokenAttr = att as Attribute;
-              });
-            } else {
-              if (mfaTokenAttr.value) {
-                const tokenStrings = mfaTokenAttr.value as string[];
-                tokenStrings.forEach(token => {
-                  const parsedToken = JSON.parse(token);
-                  this.tokens.push({
-                    added: parsedToken['added'],
-                    revoked: parsedToken['revoked'],
-                    data: parsedToken['data'],
-                    used: parsedToken['used'],
-                    type: parsedToken['type'].toUpperCase(),
-                    nickname: parsedToken['name']
-                  });
-                });
-              }
-            }
-            this.dataSource = new MatTableDataSource<MFAToken>(this.tokens);
-            this.dataSource.paginator = this.children.first.paginator;
-            this.loading = false;
-          });
-        }
-      }
-    }, () => this.loading = false);
-  }
 }
 
 export interface MFAToken {
```

Comment:
I believe this is not needed anymore along with the datasource. Otherwise it looks ok.